make-checkout

```diff
@@ -70,11 +70,24 @@ def main():
     execute("git", "checkout", "--detach", opts.revision, cwd=TARGET_DIR, error_on_fail=False)
 
     # get bots/
-    if api.repo != "cockpit-project/cockpit" or (opts.base and opts.base != "master"):
-        execute("git", "remote", "add", "bots_origin", "https://github.com/cockpit-project/cockpit", cwd=TARGET_DIR)
-        execute("git", "fetch", "--depth=1", "bots_origin", opts.bots_ref or "master", cwd=TARGET_DIR)
-        bots_ref = execute("git", "rev-parse", "FETCH_HEAD", cwd=TARGET_DIR).strip()
-        execute("git", "checkout", "--force", bots_ref or "bots_origin/master", "--", "bots/", cwd=TARGET_DIR, error_on_fail=False)
+    bots_clone_dir = os.path.join(TARGET_DIR, "bots")
+
+    # HACK
+    # As long as bots/ is duplicated in cockpit, keep the conditional section here
+    if api.repo == "cockpit-project/cockpit":
+        if opts.base and opts.base != "master":
+            execute("git", "remote", "add", "bots_origin", "https://github.com/cockpit-project/bots", cwd=TARGET_DIR)
+            execute("git", "fetch", "--depth=1", "bots_origin", opts.bots_ref or "master", cwd=TARGET_DIR)
+            bots_ref = execute("git", "rev-parse", "FETCH_HEAD", cwd=TARGET_DIR).strip()
+            execute("git", "checkout", "--force", bots_ref or "bots_origin/master", "--", "bots/", cwd=TARGET_DIR, error_on_fail=False)
+            execute("git", "reset", "bots", cwd=TARGET_DIR)
+
+    else:
+        execute("git", "clone", "--reference-if-able", "{}/{}".format(cache, "cockpit-project/bots"), "https://github.com/cockpit-project/bots", bots_clone_dir)
```

Comment:
This is already being run from a master clone of cockpit-project/bots, right? So why does this need to be fetched/checked out again into some cache dir? Couldn't this just always clone itself? (I guess the point of this new clone is to not destroy your own bots master dir with --bots-ref?)

Reply:
Well I guess we could clone it from itself, and then execute the fetch of bots_ref within that clone.

So like `git clone . make-checkout-workdir/bots` etc...

Reply:
Though if we want to have cockpit-project/cockpit still be the source of truth, we don't really need to do a clone, we can just use the same thing we've been using; a fetch --depth1 and force checkout.